solr/modules/s3-repository/src/test/org/apache/solr/s3/S3BackupRepositoryTest.java

```diff
@@ -320,17 +340,113 @@ protected URI getBaseUri() throws URISyntaxException {
+    try (S3Client s3 = createS3ClientV2()) {
       File file = temporaryFolder.newFile();
       InputStream input = s3.getObject(b -> b.bucket(BUCKET_NAME).key(path));
       FileUtils.copyInputStreamToFile(input, file);
       return file;
     }
   }
+
+  public S3Client createS3ClientV2() {
+    return S3Client.builder()
+        .region(Region.of("us-east-1"))
+        .credentialsProvider(
+            StaticCredentialsProvider.create(AwsBasicCredentials.create("foo", "bar")))
+        .endpointOverride(URI.create(S3_MOCK_RULE.getServiceEndpoint()))
+        .httpClient(
+            UrlConnectionHttpClient.builder()
+                .buildWithDefaults(
+                    AttributeMap.builder()
+                        .put(
+                            new HttpConfigurationOption("secureConnection", Boolean.class),
+                            Boolean.FALSE)
+                        .build()))
+        .build();
+  }
+
+  public ClientConfiguration configureClientToIgnoreInvalidSslCertificates(
+      final ClientConfiguration clientConfiguration) {
+
+    clientConfiguration
+        .getApacheHttpClientConfig()
+        .withSslSocketFactory(
+            new SSLConnectionSocketFactory(
+                createBlindlyTrustingSslContext(), NoopHostnameVerifier.INSTANCE));
+
+    return clientConfiguration;
+  }
+
+  private SSLContext createBlindlyTrustingSslContext() {
+    try {
+      final SSLContext sc = SSLContext.getInstance("TLS");
+
+      sc.init(
+          null,
+          new TrustManager[] {
+            new X509ExtendedTrustManager() {
+              @Override
+              public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                return null;
+              }
+
+              @Override
+              public void checkClientTrusted(
+                  final X509Certificate[] arg0, final String arg1, final Socket arg2) {
+                // no-op
+              }
+
+              @Override
+              public void checkClientTrusted(
+                  final X509Certificate[] arg0, final String arg1, final SSLEngine arg2) {
+                // no-op
+              }
+
+              @Override
+              public void checkClientTrusted(final X509Certificate[] certs, final String authType) {
+                // no-op
+              }
+
+              @Override
+              public void checkServerTrusted(final X509Certificate[] certs, final String authType) {
+                // no-op
+              }
+
+              @Override
+              public void checkServerTrusted(
+                  final X509Certificate[] arg0, final String arg1, final Socket arg2) {
+                // no-op
+              }
+
+              @Override
+              public void checkServerTrusted(
+                  final X509Certificate[] arg0, final String arg1, final SSLEngine arg2) {
+                // no-op
+              }
+            }
+          },
+          new java.security.SecureRandom());
```

Comment:
Should we use NotSecurePseudoRandom here?

Reply:
Maybe, but unrelated to the jetty changes.